odoo/addons/base/models/assetsbundle.py

```diff
@@ -348,6 +350,29 @@ def js(self):
 
         return js_attachment[0]
 
+    def xml(self):
+        if attachments := self.get_attachments('xml.js'):
+            return attachments[0]
+        bundle_content = textwrap.dedent(f"""
+            {{
+                const loadXmlTemplates = () => {{
+                    odoo.define("{self.name}.bundle.xml", ["@web/core/templates"], function (require) {{
+                        "use strict";
+                        const {{ checkPrimaryTemplateParents, registerTemplate, registerTemplateExtension }} = require("@web/core/templates");
+                        /* {self.name} */
+                        {self.generate_xml_bundle()}
+                    }});
+                }};
+                if (globalThis.odoo?.js_assets_loaded?.includes("{self.name}")) {{
+                    loadXmlTemplates();
+                }} else {{
+                    document.addEventListener("odoo:js_asset:{self.name}:loaded", loadXmlTemplates);
```

Comment:
Perhaps `js_assets_loaded` could be a set and you could dispose of its corresponding entry once the XML is loaded.